### ISSUE TITLE
Why not use AppKit/UIKit APIs for iOS/macOS targets?

### ISSUE THREAD

**Issue by colbyn** [audulus/rui#34]
It’s nice to see a UI toolkit inspired by SwiftUI, there’s a lot of things I really like about it, but I believe SwiftUI is just a VDOM-like frontend to UIKit/AppKit. 

Anyway with regards to this library, wouldn’t it make more sense to build upon UIKit/AppKit APIs for iOS and macOS targets?

It’s all written in objective-c and therefore it’s viable to use from rust, and seems like you’d get much more mileage/utility from it.

**Comment by wtholliday** [comment]
Thanks for your interest!

Yes SwiftuUI is a VDOM-like frontend to UIKit/AppKit. For Audulus, I found that CoreAnimation (which is used by both UIKit and AppKit) wasn't good enough performance, so first I used nanovg, then wrote my own (vger). Things might be better nowadays with more things rendered on the GPU, but if my recent experiments with SwiftUI's canvas are any indication, it's not very good performance (I used SwiftUI's canvas here: https://github.com/audiokit/flow). It's definitely possible that using CAShapeLayer for various things would have good enough performance for my needs, but I can't afford the time spent to find out.

In short, the way I did rui is motivated by the somewhat atypical needs of my app, Audulus. That said, a lot of other rust and non-rust UI's go the same way and don't wrap the underlying native toolkit: Qt, JUCE, Druid to name a few.

cheers

**Comment by colbyn** [comment]
@wtholliday Oh thanks for the speedy reply!
